e2eAIOK/deltatuner/deltatuner/deltatuner_model.py

```diff
@@ -218,14 +218,7 @@ def from_pretrained(
     ):
         from peft.mapping import PEFT_TYPE_TO_CONFIG_MAPPING
         from .mapping import DELTATUNER_TYPE_TO_CONFIG_MAPPING, MODEL_TYPE_TO_DELTATUNER_MODEL_MAPPING
-        denas_config = kwargs.pop("denas_config", None)
-
-        best_structure_file = os.path.join(model_id, "best_model_structure.txt")
-        if os.path.isfile(best_structure_file):
-            denas_config.denas = True
-            denas_config.best_model_structure = best_structure_file
-        else:
-            denas_config.denas = False
+        denas_config = self._get_denas_config(model_id, **kwargs)
```

Comment:
I am confused about why we need to construct the denas config here instead of inputting this by the user as we previous implementation. This implementation will limit the user's flexibility to change the deltatuner arguments.

Reply:
The reason we infer denas config from file instead of input args is as follows:
- The current logic is, in `from_pretained` function, we only support inferring denas config from file, since it should be searched already, enable user to assign the best structure need careful validation and not necessary
-  But in `optimize` function, we still support inferring denas config from input args, which is reasonable and you may not mistake this case with the above

I also double check all the args in `DeltaTunerArguments` (including `ssf_target_module` and `task_type`), all works fine with current logic.